modules/parquet/src/parquetjs/parser/parquet-reader.ts

```diff
@@ -36,7 +36,8 @@ export type ParquetIterationProps = {
  */
 export class ParquetReader {
   static defaultProps: Required<ParquetReaderProps> = {
-    defaultDictionarySize: 1e6,
+    //max ArrayBuffer size in js is 2Gb
+    defaultDictionarySize: 2147483648,
```

Comment:
That is a lot of memory... I am not sure how this value is used but I do not feel that it is acceptable to allocate that much and hold on to it. 

Also, I expect this will auto crash on mobile devices where browser tabs can only be 100-200 MB before being killed.

Is there a way we can emit a good error when it is too
Low so app knows when to raise it?

Can we adapt the value based on known length of data pages? Once the reader reads the parquet file header doesn't it know?

Can we perhaps set it to 10M? I think that is the highest reasonable default.

Reply:
We're using it as a max possible size of a dictionary, and if `this.file.size - dictionaryPageOffset` is more than `defaultDictionarySize` we use this value. So, yes, we know the correct length from the very beginning and I assume `defaultDictionarySize` is there just to provide anything but `0` or `undefined` in the case if some value if wrong.

I think we can set it up to 10 (better 50) mb and see if there is any dataset with a bigger dictionary

Also, as a separate task, we can think of any way to handle big dictionaries. Probably we can skip it somehow...

Reply:
Ok in that case I agree that we can raise the limit more.